src/patterns/security-code/index.md.njk

```diff
@@ -0,0 +1,111 @@
+---
+title: Security code
+description: Identifying users when they sign in
+section: Patterns
+theme: Ask users for…
+aliases: 2FA
+backlog_issue_id: 25
+layout: layout-pane.njk
+---
+
+{% from "_example.njk" import example %}
+
+Use this pattern to verify users via a security code sent to their phone.
+
+{{ example({group: "patterns", item: "security-code", example: "default", html: true, nunjucks: true, open: false}) }}
+
+## When to use this pattern
+
+Use this pattern if your users need to sign in to your service.
+
+Requiring users to enter a security code sent to their mobile phone provides a second layer of security over a [password](/patterns/passwords/).
+
+You can require a security code on every sign in or only once per device, depending upon the risk of your service.
+
+## How it works
+
+You should first ask users for their [password](/patterns/passwords/). Once they have successfully entered this, send a text message to their mobile phone with a 5 digit code in this format:
+
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
+{{ govukInsetText({
+  text: "12345 is your [service name] security code"
```

Comment:
GOV.UK Pay changed from this:

> **GOVUK Pay**
> 123456 is your verification code

To:
> **GOVUK Pay**
> 123456 is your verification code. Enter the code to sign in. Never tell anyone else your code. 

The reason for this change was to make it less likely that a user is tricked into disclosing the security code to someone malicious.

Is it worth recommending something similar in this pattern?

(Bolded parts represent the sender of the text message, in other words what shows instead of a phone number or phone book contact. Setting that is probably out of scope for this pattern, and would be better in the [Service Manual page on sending emails and text messages](https://www.gov.uk/service-manual/design/sending-emails-and-text-messages#when-to-send-emails-and-text-messages))

Reply:
Further to this, the text “Enter the code to sign in” can also be other things depending on the operation the user is authenticating (exhaustively, it’s “Enter the code to sign in“, “Enter it to confirm you now want to receive verification codes by text” or “Enter the code to create your account”).

The use case here is if an attacker has the user’s password and phone number. They could phone the user pretending to be GOV.UK Pay support or something, tell them they’re going to send a code they need to read back “for security purposes”, then try to sign in or whatever, triggering the code to be sent via text. The thinking was that if it says “Enter the code to sign in” and they’re not signing it, the user may realise something’s not right.

Also, “verification code” (rather than “security code” or whatever) was picked based on an unscientific analysis of what other sites called it.

Reply:
Interesting! To me, "security code" feels simpler and more Plain English than "verification code", but I've not done any specific research on this.

Reply:
Notify copied ‘security code’ from the public-facing part of GOV.UK Verify, so I imagine there was some usability done of it. I can ask around, but that would be going back to 2015/16 now.

Reply:
We used to use “security code” and “verification code” interchangeably and wanted to standardise on one.

The entirety of the conversation I can find about this decision goes as follows:

Our then tech writer: _Verification code seems to be the term mostly commonly used. When I googled security code all the results were about the numbers on the back of your credit/debit card._

Me: _I agree we should standardise on one or the other. Using the highly scientific method of looking through my text messages, “verification” seems to be slightly more common than “security” (with a few wildcards like “authentication”). Given our audience, it’s probably best to use a term that isn’t going to be confused with something on a debit/credit card. I’ll change “security code” to “verification code” everywhere it appears in the admin tool._

I’m not wedded to either term.

Reply:
For the Budgeting Loans service at DWP we chose 'security code' partly because it's a such a common term compared to alternatives (Google Trends). It's also commonly used in government services, as has been described previously.
We gave it is own name (rather than just calling it a 'code') because we needed to differentiate it from another code we gave the user in this part of the service - an application reference. Users are asked to input both to see our decision on their application. 
This is a short to medium term approach tied in with the technical architecture of the service, as there's a gradual move to digitise the entire process.

Reply:
Hi all, looking at resolving these last few threads to get this component published.

To resolve this convo on security code vs. verification code, can we go with 'security' for now and move this discussion to this component's GitHub backlog issue?

Reply:
Resolving, will add this to the backlog issue update when we publish this component.